File: src/main/java/com/lessons/leetcode/hash_table/TwoSum1.java
package com.lessons.leetcode.hash_table;


import java.util.HashMap;
import java.util.Map;

/**
 * Ready
 * <a href="https://leetcode.com/problems/two-sum/description/">...</a>
 */
public class TwoSum1 {

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numsMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            Integer find = numsMap.get(target - nums[i]);
            if (find != null)
                return new int[]{find, i};
            numsMap.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: src/main/java/com/lessons/leetcode/hash_table/TwoSum1.kt
package com.lessons.leetcode.hash_table

fun twoSum(nums: IntArray, target: Int): IntArray {
    val map = mutableMapOf<Int, Int>()
    for (i in nums.indices) {
        map[target - nums[i]]?.let { index ->
            return intArrayOf(i, index)
        }
        map[nums[i]] = i
    }
    return intArrayOf()
}
